src/pip/_internal/metadata/importlib.py

```diff
@@ -115,21 +138,21 @@ def info_location(self) -> Optional[str]:
             return None
         return str(self._info_location)
 
-    def _get_dist_name(self) -> str:
+    def _get_dist_normalized_name(self) -> NormalizedName:
         # The 'name' attribute is only available in Python 3.10 or later. We are
         # only targeting that, but Mypy does not know this.
```

Comment:
Let's add an assert here, just to make this assumption explicit?

How do you think this should interact with #10647, where there was interest in enabling sysconfig on 3.9?

Reply:
`importlib.metadata` only went out of provisional state in 3.10, and the API unstability is the main issue enabling the backend on prior to 3.10. I’m OK if someone wants to do that, but someone else would need to do the research and write the compatibility code for everything to work, and I don’t want to be that guy 🙂

Reply:
Seems reasonable to me!